gremlin-test/src/main/resources/org/apache/tinkerpop/gremlin/test/features/sideEffect/Aggregate.feature

```diff
@@ -144,145 +144,490 @@ Feature: Step - aggregate()
+    """
+    When iterated to list
+    Then the result should be unordered
+      | result   |
+      | d[124].i |
+
+
+  Scenario: g_withSideEffectXa_1_sumLongX_V_aggregateXaX_byXageX_capXaX
+    Given the empty graph
+    And the graph initializer of
+    # This graph is the same as ModernGraph, but ages are stored as Long.
+      """
+      g.addV("person").property("name", "marko").property("age", 29L).as("marko").
+        addV("person").property("name", "vadas").property("age", 27L).as("vadas").
+        addV("software").property("name", "lop").property("lang", "java").as("lop").
+        addV("person").property("name","josh").property("age", 32L).as("josh").
+        addV("software").property("name", "ripple").property("lang", "java").as("ripple").
+        addV("person").property("name", "peter").property("age", 35L).as('peter').
+        addE("knows").from("marko").to("vadas").property("weight", 0.5d).
+        addE("knows").from("marko").to("josh").property("weight", 1.0d).
+        addE("created").from("marko").to("lop").property("weight", 0.4d).
+        addE("created").from("josh").to("ripple").property("weight", 1.0d).
+        addE("created").from("josh").to("lop").property("weight", 0.4d).
+        addE("created").from("peter").to("lop").property("weight", 0.2d)
       """
-      g.V().aggregate("a").has("person","age", P.gte(30)).cap("a").unfold().values("name")
+    And the traversal of
+    """
+    g.withSideEffect("a", 1L, Operator.sumLong).V().aggregate("a").by("age").cap("a")
+    """
+    When iterated to list
+    Then the result should be unordered
+      | result   |
+      | d[124].l |
+
+  Scenario: g_withSideEffectXa_1_sumLongX_V_aggregateXlocal_aX_byXageX_capXaX
+    Given the empty graph
+    And the graph initializer of
+    # This graph is the same as ModernGraph, but ages are stored as Long.
+      """
+      g.addV("person").property("name", "marko").property("age", 29L).as("marko").
+        addV("person").property("name", "vadas").property("age", 27L).as("vadas").
+        addV("software").property("name", "lop").property("lang", "java").as("lop").
+        addV("person").property("name","josh").property("age", 32L).as("josh").
+        addV("software").property("name", "ripple").property("lang", "java").as("ripple").
+        addV("person").property("name", "peter").property("age", 35L).as('peter').
+        addE("knows").from("marko").to("vadas").property("weight", 0.5d).
+        addE("knows").from("marko").to("josh").property("weight", 1.0d).
+        addE("created").from("marko").to("lop").property("weight", 0.4d).
+        addE("created").from("josh").to("ripple").property("weight", 1.0d).
+        addE("created").from("josh").to("lop").property("weight", 0.4d).
+        addE("created").from("peter").to("lop").property("weight", 0.2d)
       """
+    And the traversal of
+    """
+    g.withSideEffect("a", 1L, Operator.sumLong).V().aggregate(Scope.local, "a").by("age").cap("a")
```

Comment:
For the test traversals, `1L` is lost in translation for GLVs, we'll need to specific this as a parameter with `d[1].l`

Reply:
Same as above.